src/main/java/core/basesyntax/TreeSetCharacters.java

```diff
@@ -17,7 +25,34 @@
  * Результат 2: acf</p>
  */
 public class TreeSetCharacters {
-    public String getUniqueCharacters(String fileName) {
-        return null;
+    public String getUniqueCharacters(String fileName) throws FileNotFoundException {
+        String line = "";
+        TreeSet<Character> treeSet = new TreeSet<>();
+        try (BufferedReader fin = new BufferedReader(new FileReader(new File(fileName)))) {
+            while ((line = fin.readLine()) != null) {
```

Comment:
try to use `reader.read()` and `reader.ready() instead of `readLine`

Reply:
Done